test/make.sh

```diff
@@ -14,63 +14,72 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# This function checks to make sure that every
-# shebang has a '- e' flag, which causes it
-# to exit on error
-function check_bash() {
-find . -name "*.sh" | while IFS= read -d '' -r file;
-do
-  if [[ "$file" != *"bash -e"* ]];
-  then
-    echo "$file is missing shebang with -e";
-    exit 1;
-  fi;
-done;
+# find_files is a helper to exclude .git directories and match only regular
+# files to avoid double-processing symlinks.
+find_files() {
+  local pth="$1"
+  shift
+  find "${pth}" -path ./.git/ -prune -o -type f "$@"
+}
+
+# Compatibility with both GNU and BSD style xargs.
+compat_xargs() {
+  local compat=()
+  if xargs --no-run-if-empty </dev/null 2>/dev/null; then
```

Comment:
```suggestion
  if xargs --help | grep --regexp '--no-run-if-empty' -; then
```

Reply:
Apologies; context would have been helpful. I was trying to provide an alternative implementation with a more human readable approach. I guess the Mac version of xargs doesn't accept `-` as an argument, but I think the GNU version doesn't require it either so it can probably be safely dropped.

Reply:
No problem!

I struggled to make this readable at first too, and try to avoid silently redirecting stderr if at all possible.

Would you approve something like this?

```bash
# Test if xargs is GNU or BSD style.  GNU xargs will succeed with status 0 when given
# --no-run-if-empty and no input on STDIN.  BSD xargs will fail and exit status non-zero
# If xargs fails, assume it is BSD style and proceed.  stderr is silently redirected to avoid
# console log spam.
if xargs --no-run-if-empty </dev/null 2>/dev/null; then
```

Reply:
:+1: